test/correctness/specialize.cpp

```diff
@@ -609,7 +609,33 @@ int main(int argc, char **argv) {
         _halide_user_assert(vector_store_lanes == 32);
     }
 
+    {
+        Var x;
+        Param<int> p;
+        Expr const_true = Expr(0) == Expr(0);
+        Expr different_const_true = Expr(1) == Expr(1);
+
+        // Check that if we promote a final const-true specialize, we keep the
+        // implicit compute/store_root required for outputs.
+        Func f("foof");
+        f(x) = x;
+        f.specialize(p == 0).vectorize(x, 32);      // will *not* be pruned
+        f.specialize(const_true);                   // dupe of call above, won't add new specialization
```

Comment:
Should vectorize by 16 in this case and check for it below so that it can be distinguished from the base case